File: 2015/02/02.go
package main

import (
	"fmt"
	"os"
	"strings"
	"strconv"
)

func main() {

	raw_data, _ := os.ReadFile("./input.txt")
	problem_input := strings.Split(string(raw_data), "\n")

	total := 0
	for _, e := range problem_input {
		inputs := strings.Split(e, "x")
		total += calculateWrappingPaper(inputs)
	}

	fmt.Println(total)

}

func calculateWrappingPaper(inputs []string) int {
	x, _ := strconv.Atoi(inputs[0])
	y, _ := strconv.Atoi(inputs[1])
	z, _ := strconv.Atoi(inputs[2])
	surfaces := []int{x*y, y*z, z*x}
	return 2*sum(surfaces) + min(surfaces)
}

func sum(arr []int) int {
	res := 0
	for _, e := range arr {
		res += e
	}
	return res
}

func min(arr []int) int {
	res := arr[0]
	for _, e := range arr {
		if e < res {
			res = e
		}
	}
	return res
}

// 0.7 ms ± 0.6 ms (10k runs, 1k warmup) - Go 1.19: build

File: 2015/02/02.rs
use std::fs;

fn main() {

    let contents = fs::read_to_string("./input.txt")
        .expect("Something went wrong reading the file");
    let vec_inputs = contents.split("\n").collect::<Vec<&str>>();

    let mut acc = 0;
    for el in vec_inputs {

        let vec_row = el.split("x").collect::<Vec<&str>>();
        let a: i32 = vec_row[0].parse().unwrap();
        let b: i32 = vec_row[1].parse().unwrap();
        let c: i32 = vec_row[2].parse().unwrap();

        let res = calculate_wrapping_paper(a, b, c);
        acc += res;
    }
    println!("{}", acc);
}

fn calculate_wrapping_paper(a: i32, b: i32, c: i32) -> i32 {
    let surfaces = [a*b, b*c, c*a];
    let min = surfaces.iter().min().unwrap();
    let sum: i32 = surfaces.iter().sum();
    return 2*sum + min
}

// 0.5 ms ± 0.6 ms (10k runs, 1k warmup) - Rustc 1.61.0: -opt-level=3